overops-functions/src/main/java/com/takipi/udf/jira/JiraIntegrationFunction.java

```diff
@@ -0,0 +1,236 @@
+		List<EventResult> events = eventsResult.events;
+
+		// for each event with a Jira issue URL
+		for (EventResult event : events) {
+			if (event.jira_issue_url != null) {
+				String issueId = getJiraIssueId(event.jira_issue_url);
+				eventList.addEvent(issueId, event);
+			}
+		}
+
+		return eventList;
+	}
+
+	// get Jira issue ID from Jira issue URL
+	private static String getJiraIssueId(String jiraURL) {
+		int index = jiraURL.lastIndexOf("/")+1;
+		return jiraURL.substring(index);
+	}
+
+	private static JiraIntegrationInput getJiraIntegrationInput(String rawInput) {
+		// params cannot be empty
+		if (Strings.isNullOrEmpty(rawInput))
+			throw new IllegalArgumentException("Input is empty");
+
+		JiraIntegrationInput input;
+
+		// parse params
+		try {
+			input = JiraIntegrationInput.of(rawInput);
+		} catch (Exception e) {
+			throw new IllegalArgumentException(e.getMessage(), e);
+		}
+
+		// validate days
+		if (input.days <= 0)
+			throw new IllegalArgumentException("'days' must be at least 1 day");
+
+		if (input.jiraURL == null || input.jiraURL.isEmpty())
+			throw new IllegalArgumentException("'jiraURL' is required");
+
+		if (input.jiraUsername == null || input.jiraUsername.isEmpty())
+			throw new IllegalArgumentException("'jiraUsername' is required");
+
+		if (input.jiraPassword == null || input.jiraPassword.isEmpty())
+			throw new IllegalArgumentException("'jiraPassword' is required");
+
+		if (input.resolvedStatus == null || input.resolvedStatus.isEmpty())
+			throw new IllegalArgumentException("'resolvedStatus' is required");
+
+		if (input.hiddenStatus == null || input.hiddenStatus.isEmpty())
+			throw new IllegalArgumentException("'hiddenStatus' is required");
+
+		return input;
+	}
+
+	static class JiraIntegrationInput extends Input {
+		public int days; // in days
+
+		public String jiraURL;
+		public String jiraUsername;
+		public String jiraPassword;
+
+		public String resolvedStatus;
+		public String hiddenStatus;
+
+		private JiraIntegrationInput(String raw) {
+			super(raw);
+		}
+
+		@Override
+		public String toString() {
+			StringBuilder builder = new StringBuilder();
+
+			builder.append("Sync Jira (");
+			builder.append(days);
+			builder.append("d)");
+
+			return builder.toString();
+		}
+
+		static JiraIntegrationInput of(String raw) {
+			return new JiraIntegrationInput(raw);
+		}
+	}
+
+	// for testing in aws lambda
+	public static void lambdaHandler(String input, Context context) {
+		System.out.println("lambdaHandler()");
+		System.out.println(input);
+		main(input.split(" "));
+	}
+
+	// for testing locally
+	public static void main(String[] args) {
+		Instant start = Instant.now(); // timer
+
+		if ((args == null) || (args.length < 9))
+			throw new IllegalArgumentException(
+				"java JiraIntegrationFunction API_URL API_KEY SERVICE_ID JIRA_URL JIRA_USER JIRA_PASS " + 
+				"DAYS RESOLVED_STATUS HIDDEN_STATUS");
+
+		ContextArgs contextArgs = new ContextArgs();
+
+		contextArgs.apiHost = args[0];
```

Comment:
You can easily create context args string via `TestUtil.getViewContextArgs`

Reply:
awesome! didn't know about `TestUtil`